AsyncDisplayKit/ASImageNode.mm

```diff
@@ -134,11 +127,11 @@ - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
 
 - (void)setImage:(UIImage *)image
 {
-  _imageLock.lock();
```

Comment:
I actually remember now why this method had to use .lock() and .unlock() manually.  It was to try to avoid interleaving with the superclass _propertyLock, which was grabbed inside invalidateCalculatedLayout and other methods below.

I think now that this is addressed, we should make this case a simple MutexLocker.  The else case that unlocks can be deleted.